### DISCUSSION TITLE
Elegoo Mars 4 Ultra .goo format for Phased Exposure

### DISCUSSION THREAD

**Discussion by research-e** [sn4k3/UVtools#922] [Q&A] [answered]
Hi, I have tried printing with the phased exposure for the Mars 4 Ultra for phased exposure. The resultant prints produce layer heights equal to layer height *num of exposures so with 5 exposures at 50um layers its printing 250um layers. Is this an issue with the .goo format? I was using lychee slicer and it offers both .goo and .ctb formats. I have attached the sliced file if that helps.

[Elegoo Exposure finder.zip](https://github.com/user-attachments/files/16801529/Elegoo.Exposure.finder.zip)

I will also give the .ctb file a test print. to see if its a general compatibility issue

**Comment by sn4k3** [comment, marked as answer]
Goo format can accept different parameters per layer, however it all depends on printer firmware, some looks not to accept that. Same goes with CTB some printers can while others don't. There also cases where people updated firmware and suddenly printer no longer able to run these when before they was.
This is really a mess and again mSLA is going down the pit with this bad firmwares...

**Reply by research-e** [reply]
ah okay makes sense, if i find something which works i can record firmware and file type for compatibility records. 
Shame about the mess with firmwares, has so much potential

**Reply by research-e** [reply]
I can confirm changing slicer file to .ctb format makes no difference. Weirdly enough its enough to get by if phased exposure is only needed for a handful of layers not too close to eachother. You can get multi layers at the same height with 0 bed axis movement and then the next layer after it moves (when you are done with phased exposure for that section),  it will essentially move to the height equal to layer height*number of layers

**Reply by sn4k3** [reply]
The firmwares are bugwares, and we can never know what's udner the hood.
If you want free and full control consider a convert to klipper if your printer has HDMI

**Reply by research-e** [reply]
I had not heard about that until now. I will definitely take a look, thanks!
